test/test_kubeclient.rb

```diff
@@ -92,13 +92,31 @@ def test_api_valid
     stub_request(:get, 'http://localhost:8080/api')
       .to_return(status: 200, body: open_test_json_file('versions_list.json'))
 
+    %w(v1beta1 v1beta2 v1beta3).each do |version|
+      client = Kubeclient::Client.new 'http://localhost:8080/api/', version
+      assert client.api_valid?
+    end
+  end
+
+  def test_api_valid_with_invalid_version
+    stub_request(:get, 'http://localhost:8080/api')
+      .to_return(status: 200, body: open_test_json_file('versions_list.json'))
+
+    client = Kubeclient::Client.new 'http://localhost:8080/api/', 'foobar1'
+    refute client.api_valid?
+  end
+
+  def test_api_valid_with_missing_version
```

Comment:
the method name is not very clear until looking at the code.
I'd say it's more missing_versions_block kind of thing :)
missing_version implies to me that there's one specific version missing from the response...